lib/fika/env.ex

```diff
@@ -10,26 +10,26 @@ defmodule Fika.Env do
       # This is a map of known function signatures and their return types.
       # As a hack for now, we store kernel function signatures right here.
       function_types: %{
-        "kernel.+(Int,Int)" => "Int",
-        "kernel.+(Int,Float)" => "Float",
-        "kernel.+(Float,Int)" => "Float",
-        "kernel.+(Float,Float)" => "Float",
-        "kernel.-(Int,Int)" => "Int",
-        "kernel.-(Int,Float)" => "Float",
-        "kernel.-(Float,Int)" => "Float",
-        "kernel.-(Float,Float)" => "Float",
-        "kernel.*(Int,Int)" => "Int",
-        "kernel.*(Int,Float)" => "Float",
-        "kernel.*(Float,Int)" => "Float",
-        "kernel.*(Float,Float)" => "Float",
-        "kernel./(Int,Int)" => "Float",
-        "kernel./(Int,Float)" => "Float",
-        "kernel./(Float,Int)" => "Float",
-        "kernel./(Float,Float)" => "Float",
-        "kernel.|(Bool,Bool)" => "Bool",
-        "kernel.&(Bool,Bool)" => "Bool",
-        "kernel.!(Bool)" => "Bool",
-        "kernel.-(Int)" => "Int"
+        "kernel.+(Int,Int)" => :Int,
```

Comment:
I'm not sure about this change. I'd only keep this if we choose to parse the written typespecs in code as their higher-order counterparts as opposed to their string-values